gobblin-utility/src/main/java/org/apache/gobblin/util/jdbc/DataSourceProvider.java

```diff
@@ -81,6 +81,8 @@ public DataSourceProvider(@Named("dataSourceProperties") Properties properties)
   }
 
   public DataSourceProvider() {
+    LOG.warn("Creating {} without setting validation query. Stacktrace of current thread {}",
+        this.getClass().getSimpleName(), Thread.currentThread().getStackTrace());
```

Comment:
because this is a `StackTraceElement[]`, `.toString` will be ugly.  you may want something like:
```
Arrays.toString(st).replace(", ", "\n  at ")
```
or
```
String.join("\n  at", st.stream().map(x -> x.toString()).collect(...))
```